esmvaltool/interface_scripts/logging.ncl

```diff
@@ -334,9 +334,38 @@ begin
 
   ; Save entries for outfile if not "n/a"
   if (outfile .ne. "n/a") then
-    outstring2 = outstring
-    outstring2(0) = "? " + outfile
-    outstring := array_append_record(outstring, outstring2, 0)
+    suffix = get_file_suffix(outfile, 0)
+
+    ; For PNGs, additionally check for existence of files like
+    ; "plot_file.000001.png", "plot_file.000002.png", etc. and save
+    ; provenance record for each of them
+    if ((suffix .eq. ".png") .and. (.not. fileexists(outfile))) then
+      do file_idx = 1, 999999
+        potential_outfile = suffix@fBase + "." + sprinti("%0.6i", file_idx) + \
+                            suffix
+        if (fileexists(potential_outfile)) then
+          if (.not. isvar("all_outfiles")) then
+            all_outfiles = potential_outfile
+          else
+            all_outfiles := array_append_record(all_outfiles, \
+                                                potential_outfile, 0)
+          end if
+        else
+          break
+        end if
+      end do
+    else
+      all_outfiles = outfile
+    end if
+
+    ; Save provenance record of all files
+    original_entry = outstring
+    do outfile_idx = 0, dimsizes(all_outfiles) - 1
+      file_to_add := all_outfiles(outfile_idx)
+      new_entry = original_entry
```

Comment:
In order to avoid a warning about non-matching dimension names, I suggest this change:
```suggestion
      new_entry = (/original_entry/)
```